### PR TITLE
executor: skip execution when build query for VIEW in I_S (#58203)

### LINKED ISSUE #58184: `INFORMATION_SCHEMA.columns` performance regression after fix #54343
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
use test;
drop table if exists t;
create table t(a int);
insert into t values(1);
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
insert into t select * from t;
create view v as select * from t where a = (select max(t2.a) from t t2);
create view v2 as select * from t where a = (select max(t2.a) from t t2);
create view v3 as select * from t where a = (select max(t2.a) from t t2);
create view v4 as select * from t where a = (select max(t2.a) from t t2);
create view v5 as select * from t where a = (select max(t2.a) from t t2);
create view v6 as select * from t where a = (select max(t2.a) from t t2);
select count(1) from INFORMATION_SCHEMA.columns;
```

thanks for @wshwsh12

### 2. What did you expect to see? (Required)

before the fix, there's some warnings but the execution time is small

use v6.5.8

```
mysql> select count(1) from INFORMATION_SCHEMA.columns;
+----------+
| count(1) |
+----------+
|     4629 |
+----------+
1 row in set, 6 warnings (0.03 sec)
```

### 3. What did you see instead (Required)

after the fix, execution is slow

use nightly (~8.6)

```
mysql> select count(1) from INFORMATION_SCHEMA.columns;
+----------+
| count(1) |
+----------+
|     4981 |
+----------+
1 row in set (0.74 sec)
```

use v6.5.11

```
mysql> select count(1) from INFORMATION_SCHEMA.columns;
+----------+
| count(1) |
+----------+
|     4635 |
+----------+
1 row in set (0.66 sec)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #58184

Problem Summary:

### What changed and how does it work?

use `PlanBuilderOptNoExecution` to skip unnecessary execution. This should be the same behaviour with `SHOW COLUMNS`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

File: tools/check/check-bazel-prepare.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #  -o pipefail: sets the exit code of a pipeline to that of the rightmost command to exit with a non-zero status,
 #       or to zero if all commands of the pipeline exit successfully.
 set -euo pipefail
+rm -rf /home/jenkins/.cache/bazel/_bazel_jenkins/install/a09dbb90c658248f08f9aa0eba11997d
```
Comment (tangenta) [inline tools/check/check-bazel-prepare.sh:22]: Could you explain this change?

**Comment by tangenta** [comment]
/hold 

I have a comment, but feel free to unhold this

File: tools/check/check-bazel-prepare.sh (hunk 1 above)
Comment (lance6716) [reply tools/check/check-bazel-prepare.sh:22]: bazel building system leaves an invalid cache in our CI, need to manually delete it. Will send you details in our IM

**Pull request merged**